test/src/main/java/io/zeebe/test/exporter/MockScheduledTask.java

```diff
@@ -52,4 +70,37 @@ public void run() {
   public boolean wasExecuted() {
     return executed;
   }
+
+  @Override
+  public void cancel() {
+    if (!isCanceled()) {
+      task.cancel();
+      canceled = true;
+    }
+  }
+
+  public boolean isCanceled() {
+    return canceled;
+  }
+
+  interface ScheduledRunnable extends ScheduledTask, Runnable {}
+
+  private static final class DefaultScheduledTask implements ScheduledRunnable {
```

Comment:
Let's keep it simple. I don't think these additional interface and class is required. Instead you can modify the run method as follows:
```
public void run() {
    if (!wasExecuted() && !isCancelled()) {
      task.run();
      executed = true;
    }
  }
```

Reply:
Done